testcrate/ui/not-a-result.stderr

```diff
@@ -18,14 +37,49 @@ error[E0277]: the trait bound `u32: futures::__rt::IsResult` is not satisfied
    |
    = help: the trait `futures::__rt::IsResult` is not implemented for `u32`
 
+error[E0308]: mismatched types
+  --> $DIR/not-a-result.rs:12:1
+   |
+12 | #[async(boxed)]
+   | ^^^^^^^^^^^^^^^ expected enum `std::result::Result`, found u32
+   |
+   = note: expected type `std::result::Result<_, _>`
+              found type `u32`
+
 error[E0277]: the trait bound `u32: futures::__rt::IsResult` is not satisfied
   --> $DIR/not-a-result.rs:13:13
    |
 13 | fn bar() -> u32 {
    |             ^^^ async functions must return a `Result` or a typedef of `Result`
    |
    = help: the trait `futures::__rt::IsResult` is not implemented for `u32`
-   = note: required by `futures::__rt::gen`
+
+error[E0308]: mismatched types
+  --> $DIR/not-a-result.rs:14:5
+   |
+14 |     3
+   |     ^ expected enum `std::result::Result`, found integral variable
+   |
+   = note: expected type `std::result::Result<_, _>`
+              found type `{integer}`
+
+error[E0308]: mismatched types
+  --> $DIR/not-a-result.rs:17:1
+   |
+17 | #[async_stream(item = u32)]
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `std::result::Result`, found u32
+   |
+   = note: expected type `std::result::Result<(), _>`
+              found type `u32`
+
```

Comment:
I think this can be confusing. ( `-> u32` is wrong, but error points `item = u32` )